File: Codeforces rounds/Codeforces Beta Round 4 (Div. 2 Only)/C.cpp
// https://codeforces.com/contest/4/problem/C

#include <bits/stdc++.h>

using namespace std;

typedef unsigned int uint;

int main(void) {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);

  uint n;
  cin >> n;

  map<string, uint> m;

  while (n--) {
    string name;
    cin >> name;
    m[name]++;

    uint count = m[name];

    if (count == 1) {
      cout << "OK\n";
    } else {
      cout << name << count - 1 << '\n';
    }
  }

}

File: Codeforces rounds/Codeforces Beta Round 4 (Div. 2 Only)/C.py
# https://codeforces.com/contest/4/problem/C

def main():

    n: int = int(input())

    m: dict[str, int] = {}

    for _ in range(n):
        name: str = input()
        if name in m:
            print(name + str(m[name]))
            m[name] += 1
        else:
            print('OK')
            m[name] = 1

if __name__ == '__main__':
    main()
